### PR TITLE
scripts: add constraint to file readable test

### PR DESCRIPTION
**Pull request by thc202**
Do not run the test if not able to change the readable permission and use newer APIs to check/change it.

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]
These all related to windows tests?

**Comment by thc202** [comment]
Yes.

File: addOns/scripts/src/test/java/org/zaproxy/zap/extension/scripts/automation/ScriptJobUnitTest.java [hunk 1]
```diff
@@ -676,6 +682,16 @@ void shouldFailToAddIfFileNotReadable() throws IOException {
         assertThat(progress.getErrors(), contains("!scripts.automation.error.file.cannotRead!"));
     }
 
+    private static void setNotReadable(Path file) throws IOException {
+        assumeTrue(
+                Files.getFileStore(file).supportsFileAttributeView(PosixFileAttributeView.class),
+                "Test requires support for POSIX file attributes.");
+        Set<PosixFilePermission> perms =
+                Files.readAttributes(file, PosixFileAttributes.class).permissions();
+        perms.remove(PosixFilePermission.OWNER_READ);
```
Comment (ricekot) [inline addOns/scripts/src/test/java/org/zaproxy/zap/extension/scripts/automation/ScriptJobUnitTest.java:691]: Do we need to remove `GROUP_READ` and `OTHERS_READ` as well?

Comment (thc202) [reply addOns/scripts/src/test/java/org/zaproxy/zap/extension/scripts/automation/ScriptJobUnitTest.java:691]: It's enough to remove `OWNER_READ` for the test.

**Review by ricekot — APPROVED** [review]
Thanks!

**Pull request merged**